core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/authorization/JDBCAuthorizationManager.java

```diff
@@ -1086,9 +1088,17 @@ private void addAuthorizationForRole(String roleName, String resourceId, String
                             + " of domain: " + domain + " to resource: " + resourceId);
                 }
 
-                DatabaseUtil.updateDatabase(dbConnection, DBConstants.ADD_ROLE_PERMISSION_SQL, permissionId,
-                        UserCoreUtil.removeDomainFromName(roleName), allow, tenantId, tenantId,
-                        domain);
+                try {
+                    DatabaseUtil.updateDatabase(dbConnection, DBConstants.ADD_ROLE_PERMISSION_SQL, permissionId,
+                            UserCoreUtil.removeDomainFromName(roleName), allow, tenantId, tenantId,
```

Comment:
Better to check why did we pass tenant id twice here as parameters (Even though it was here in old implementation as well)